code/game/objects/items/weapons/RCD.dm

```diff
@@ -419,12 +427,14 @@ obj/item/weapon/construction
 	delay_mod = 0.6
 	ranged = TRUE
 	icon_state = "arcd"
+	item_state = "rcd"
 
 /obj/item/weapon/construction/rcd/arcd/afterattack(atom/A, mob/user)
-	range_check(A,user)
+	if(!range_check(A,user))
+		return
 	if(target_check(A,user))
 		user.Beam(A,icon_state="rped_upgrade",time=30)
```

Comment:
I asked around but there really isn't a great way to do it, you can try asking again - it seems like it would be similar to "do_after"